### PR TITLE
NS-77: Update CAPI/CAPO

### PR DESCRIPTION
**Pull request by haringsrob**

### CODE REVIEW

File: charts/cluster-api-provider-openstack/templates/clusterrole-0.yaml [hunk 1]
```diff
@@ -74,6 +74,26 @@ rules:
   - get
   - patch
   - update
+- apiGroups:
```
Comment (spjmurray) [inline charts/cluster-api-provider-openstack/templates/clusterrole-0.yaml:77]: Note to self: what is this they have added?  Sounds interesting.

File: charts/cluster-api-provider-openstack/templates/validatingwebhookconfiguration-0.yaml [hunk 2]
```diff
@@ -13,15 +13,15 @@ webhooks:
     service:
       name: capo-webhook-service
       namespace: capo-system
-      path: /validate-infrastructure-cluster-x-k8s-io-v1alpha7-openstackcluster
+      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-openstackcluster
```
Comment (spjmurray) [inline charts/cluster-api-provider-openstack/templates/validatingwebhookconfiguration-0.yaml:16]: Note to self: A->B sounds fraught with peril, nvesigate

**Review by spjmurray — COMMENTED** [review]
Just a couple things needs investigating from a due diligence perspective first...

File: charts/cluster-api-provider-openstack/templates/validatingwebhookconfiguration-0.yaml (hunk 2 above)
Comment (spjmurray) [reply charts/cluster-api-provider-openstack/templates/validatingwebhookconfiguration-0.yaml:16]: Okay, this doesn't make sense, this does not exist in the '0.9.0' tag (and neither do the v1beta1 CRDs, which is my concern), but does on 'main'.  Someone (not you) has fucked something up!

Comment (spjmurray) [reply charts/cluster-api-provider-openstack/templates/validatingwebhookconfiguration-0.yaml:16]: Confusing myself evidently 😸  Mondays!

**Comment by spjmurray** [comment]
Okay, from the release notes: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/releases/tag/v0.10.0

> v1alpha6 and v1alpha7 objects will be automatically converted to v1beta during use. This is well tested. We don’t anticipate problems with these conversions.

This will create problems in that the `cluster-api-cluster-openstack` chart will create resources for v1alpha7, CAPO will convert them to v1beta1, then ArgoCD will revert them back to v1alpha7, and so on forever.  You should actually see this happening in the  ArgoCD UI.

Fix here should be easy enough, `find charts/cluster-api-cluster-openstack -type f -name -exec -i sed 's/v1alpha7/v1beta1/g' {} \;`

However, it gets more complicated in that this shit is broken, so we need some hacks to manually remove things because CAPO blocks Argo from doing so, and Argo haven't accepted any of our patches yet 🙄  I appear to have even missed this last time around too https://github.com/unikorn-cloud/unikorn/blob/main/pkg/provisioners/helmapplications/clusteropenstack/hack.go#L108 (the shame!), but it will break eventually when the version disappears.

Testing should look like:

* Can I create a cluster?
** Does Argo constantly flip flop?  If so fix it
* Can I remove a workload pool and create a new one with a different name (test the hack code)?
* Can I delete the cluster?

**Comment by yankcrime** [comment]
Closing in favour of #4

**Pull request closed without merging**